Library/Homebrew/env_config.rb

```diff
@@ -515,7 +515,13 @@ def env_method_name(env, hash)
 
       if hash[:boolean]
         define_method(method_name) do
-          ENV[env].present?
+          env_value = ENV.fetch(env, nil)
+
+          falsy_values = %w[false no off nil 0]
+          odeprecated "#{env}=#{env_value}", "#{env}=1" if falsy_values.include?(env_value&.downcase)
```

Comment:
```suggestion
          odeprecated "#{env}=#{env_value}", "#{env}=1 or unset #{env}" if falsy_values.include?(env_value&.downcase)
```

maybe even reversing or just specifying the `unset` as it might not be clear otherwise to someone attempting to disable this which option they should pick?